src/core/qgsfeaturerequest.h

```diff
@@ -175,6 +233,30 @@ class CORE_EXPORT QgsFeatureRequest
      */
     QgsFeatureRequest& disableFilter() { mFilter = FilterNone; return *this; }
 
+    /**
+     * Adds a new OrderByClause
+     *
+     * @param expression The expression to use for ordering
+     * @param ascending  If the order should be ascending (1,2,3) or descending (3,2,1)
+     *                   If thr order is ascending, by default nulls are last
+     *                   If thr order is descending, by default nulls are first
+     */
+
+    QgsFeatureRequest& addOrderBy( const QString &expression, bool ascending = true );
```

Comment:
might be worth noting in the docs that calling this multiple times will add additional order by clauses, not remove the existing ones